controllers/login.js

```diff
@@ -0,0 +1,75 @@
+require('dotenv').config();
+let jwt = require('jsonwebtoken');
+var basicAuth = require('basic-auth');
+var Customers = require('../models').customer;
+var bcrypt = require('bcrypt');
+
+class LoginController {
+  static async login(req, res) {
+    var user = basicAuth(req);
+    if (!user || !user.name || !user.pass) {
+      res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
+      res.status(401).json({
+        message: 'Failed to authorize user. Kindly check your credentials!!'
+      });
+      return;
+    }
+    if (user.name && user.pass) {
+      await Customers.findOne({
```

Comment:
I suggest
const customer = await Customers.findOne({ where: { name: user.name } });
if (customer && !Object.keys(customer.lenght)) {
return res.status(400).json({
        success: false,
        message: 'Authentication failed! Please check the request'
      });
}
 const result = await bcrypt.compare(user.pass, customer.password);
if (!result) {
          res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
           return  res.status(401).json({
                    message:
                      'Failed to authorize user. Kindly check your credentials!!'
                  });
}           
     let token = jwt.sign({ username: user.name },process.env.SECRET,
                    {
                      expiresIn: '24h' // expires in 24 hours
                    }
                  );
                  res.json({
                    success: true,
                    message: 'Authentication successful!',
                    token: token
                  });
                }

Reply:
You can use asyn await in this way no need to nest inside the then statements.